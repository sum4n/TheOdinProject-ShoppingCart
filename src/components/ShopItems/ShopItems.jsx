import styles from "./ShopItems.module.css";

const ShopItems = ({ title, description, rating, price, imgSource }) => {
  return (
    <div className={styles.shopItem}>
      <img className={styles.itemImg} src={imgSource} alt="This is shop item" />
      <div>
        <p className={styles.itemTitle}>{title}</p>
        <p>{description}</p>
        <p>
          {rating.rate} of {rating.count}
        </p>
        <p className={styles.priceText}>Price: {price}$</p>
        <div>
          <label className={styles.inputLabel} htmlFor="quantity">
            Quantity
          </label>
          <input
            className={styles.inputBox}
            type="number"
            id="quantity"
            name="quantity"
            max={10}
            min={1}
            defaultValue={1}
          />
        </div>
        <button>Add to cart</button>
      </div>
    </div>
  );
};

export default ShopItems;
