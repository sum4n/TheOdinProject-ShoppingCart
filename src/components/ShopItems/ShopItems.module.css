.shopItem {
  /* border: 2px solid black; */
  border-radius: 4px;

  display: flex;
  /* flex-direction: column; */
  flex-wrap: nowrap;
  /* justify-content: space-between; */
  gap: 8px;
  /* align-items: center; */

  /* height: 500px; */
  /* width: 200px; */

  padding-bottom: 4px;

  box-shadow: 0px 0px 8px rgb(0, 0, 0);
}

.itemTitle {
  margin: 0;
}

.itemImg {
  height: 200px;
  width: 200px;
}

.inputLabel {
  font-size: small;
  font-weight: bold;
}

.inputBox {
  width: 40px;
}
