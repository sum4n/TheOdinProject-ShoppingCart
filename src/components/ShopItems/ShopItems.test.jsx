import { render } from "@testing-library/react";
import { describe, expect } from "vitest";
import ShopItems from "./ShopItems";

describe("ShopItems component", () => {
  it("Renders with prop data", () => {
    const { container } = render(
      <ShopItems
        title={"Product 1"}
        index={1}
        description={"A product"}
        rating={{ rate: 3, count: 33 }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
