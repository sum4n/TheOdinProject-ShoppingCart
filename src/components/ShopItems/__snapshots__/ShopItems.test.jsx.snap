// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShopItems component > Renders with prop data 1`] = `
<div>
  <div
    class="_shopItem_3f94c9"
  >
    <img
      alt="This is shop item"
      class="_itemImg_3f94c9"
    />
    <div>
      <p
        class="_itemTitle_3f94c9"
      >
        Product 1
      </p>
      <p>
        A product
      </p>
      <p>
        3
         of 
        33
      </p>
      <p
        class="_priceText_3f94c9"
      >
        Price: 
        $
      </p>
      <div>
        <label
          class="_inputLabel_3f94c9"
          for="quantity"
        >
          Quantity
        </label>
        <input
          class="_inputBox_3f94c9"
          id="quantity"
          max="10"
          min="1"
          name="quantity"
          type="number"
          value="1"
        />
      </div>
      <button
        id="1"
      >
        Add to cart
      </button>
    </div>
  </div>
</div>
`;
